contracts/Everlong.sol

```diff
@@ -66,64 +67,313 @@ import { EVERLONG_KIND, EVERLONG_VERSION } from "./libraries/Constants.sol";
 /// @custom:disclaimer The language used in this code is for coding convenience
 ///                    only, and is not intended to, and does not, have any
 ///                    particular legal or regulatory significance.
-contract Everlong is EverlongAdmin, EverlongPositions, EverlongERC4626 {
+contract Everlong is IEverlong {
+    using FixedPointMath for uint256;
+    using HyperdriveExecutionLibrary for IHyperdrive;
+    using Portfolio for Portfolio.State;
+    using SafeCast for *;
+
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Storage                                                 │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    // ───────────────────────── Immutables ──────────────────────
+
+    // NOTE: Immutables accessed during transactions are left as internal
+    //       to avoid the gas overhead of auto-generated getter functions.
+    //       https://zokyo-auditing-tutorials.gitbook.io/zokyo-gas-savings/tutorials/gas-saving-technique-23-public-to-private-constants
```

Comment:
This is interesting. I read through this link, and I actually wrote a test locally to see if there was overhead to accessing `internal` vs `public` immutables. The idea that the cost would be different was surprising to me since we shouldn't be using the public getter when accessing them internally. I didn't find the gas costs to be any different, but I could be missing something.

Here is the `contracts/src/Example.sol`:

```
// SPDX-License-Identifier: Apache-2.0
pragma solidity 0.8.22;

contract ExampleA {
    uint256 public immutable im;

    constructor(uint256 _im) {
        im = _im;
    }

    function add(uint256 _value) external view returns (uint256) {
        return im + _value;
    }
}

contract ExampleB {
    uint256 internal immutable im;

    constructor(uint256 _im) {
        im = _im;
    }

    function add(uint256 _value) external view returns (uint256) {
        return im + _value;
    }
}
```

Here's `test/Example.t.sol`:

```
// SPDX-License-Identifier: Apache-2.0
pragma solidity 0.8.22;

import { console2 as console } from "forge-std/console2.sol";
import { Test } from "forge-std/Test.sol";
import { ExampleA, ExampleB } from "contracts/src/Example.sol";

contract ExampleTest is Test {
    ExampleA internal exampleA;
    ExampleB internal exampleB;

    function setUp() external {
        exampleA = new ExampleA(1);
        exampleB = new ExampleB(1);
    }

    function test_example_a() external {
        uint256 gasBefore = gasleft();
        exampleA.add(10);
        console.log("gas used = %s", gasBefore - gasleft());
    }

    function test_example_b() external {
        uint256 gasBefore = gasleft();
        exampleB.add(10);
        console.log("gas used = %s", gasBefore - gasleft());
    }
}
```

The output when running both tests was:

```
Ran 2 tests for test/Example.t.sol:ExampleTest
[PASS] test_example_a() (gas: 8851)
Logs:
  gas used = 5386

[PASS] test_example_b() (gas: 8829)
Logs:
  gas used = 5386

Suite result: ok. 2 passed; 0 failed; 0 skipped; finished in 5.88ms (1.44ms CPU time)
```

Reply:
To be clear, I would definitely expect deployment costs to be higher with the public getters, but assuming that we're going to expose these getters elsewhere, there shouldn't be a difference. The question I was trying to answer was whether or not internal accesses would be more expensive.

Reply:
Ahh when I had tested this I just looked at the testing function gas costs w/o checking the gas left. 

I guess still worth keeping for the lower deployment costs?

Reply:
I don’t think so. It’s good practice for all immutables to have getters for better UX and DevEx (this helps with etching), and it seemed like most (all?) of these already had getters elsewhere. With that said, the cost will be the same, it’s just more verbose

Reply:
okay IIRC will need to leave some as internal to play nicely with solady erc4626 (don't see a need for us to reimplement and should speed up audit times not doing so), but bulk majority should be fine to make public

Reply:
Yep, makes sense